DNN Platform/Library/Services/Installer/Installers/ScriptInstaller.cs

```diff
@@ -267,18 +317,30 @@ protected override void ProcessFile(InstallFile file, XPathNavigator nav)
                     // This is the initial script when installing
                     this._installScript = file;
                 }
-                else if (file.Name.StartsWith("upgrade.", StringComparison.InvariantCultureIgnoreCase))
+                else if (file.Name.StartsWith("preupgrade.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("preupgrade", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    this._preUpgradeScript = file;
+                }
+                else if (file.Name.StartsWith("upgrade.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("postupgrade", StringComparison.InvariantCultureIgnoreCase))
                 {
                     this._upgradeScript = file;
                 }
+                else if (file.Name.StartsWith("schema.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("schema", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    this._schemaScripts.Add(file);
+                }
                 else if (type.Equals("install", StringComparison.InvariantCultureIgnoreCase))
                 {
                     // These are the Install/Upgrade scripts
                     this.InstallScripts[file.Version] = file;
                 }
-                else
+                else if (file.Name.StartsWith("uninstall.", StringComparison.InvariantCultureIgnoreCase)

```

Comment:
The condition on [line 360](https://github.com/dnnsoftware/Dnn.Platform/blob/303a8346cf937915d5c38dcb5a1d91948b281add/DNN%20Platform/Library/Services/Installer/Installers/ScriptInstaller.cs#L360) needs to be similarly adjusted.  It seems like that condition is superfluous, since we're also checking if the file is in `this.UnInstallScripts`, so maybe the `if` on line 360 could be simplied to just be `if (this.IsValidScript(fileExtension))`

Reply:
Correct. And while we're at it, why is the check IsValidScript here and not in the logic which parses out the scripts in the first place?

Reply:
Done